File: app/src/main/java/com/example/mytestapplication/HomepageActivity.java
package com.example.mytestapplication;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;

public class HomepageActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // Load the homepage.xml layout
        setContentView(R.layout.homepage); // Make sure this layout exists in res/layout
    }
}

File: app/src/main/java/com/example/mytestapplication/HomepageActivity.kt
package com.example.mytestapplication

import android.content.Intent
import android.os.Bundle
import android.widget.ImageButton
import androidx.appcompat.app.AppCompatActivity

class HomePageActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        // homepage
        setContentView(R.layout.homepage)
        // declare button
        val settingsButton: ImageButton = findViewById(R.id.settingsbutton)

        // onclick,
        settingsButton.setOnClickListener {
            // go to settings
            val intent = Intent(this, SettingsActivity::class.java)
            startActivity(intent)
        }
    }
}
